pkg/steps/testing.go

```diff
@@ -37,13 +76,97 @@ func (dl *DryLogger) AddObject(o runtime.Object) {
 	dl.objects = append(dl.objects, o)
 }
 
-// GetObjects returns the list of objects.
+func (dl *DryLogger) AddBuild(build *buildapi.Build) {
+	imageLabels := build.Spec.Output.ImageLabels
+	if len(imageLabels) > 0 {
+		sort.Slice(imageLabels, func(i, j int) bool {
+			return imageLabels[i].Name < imageLabels[j].Name
+		})
+	}
+
+	if images := build.Spec.Source.Images; len(images) > 0 {
+		for i, image := range images {
+			build.Spec.Source.Images[i].From.Name = trimSHA256(image.From.Name)
+		}
+
+	}
+
+	dl.AddObject(build.DeepCopyObject())
+}
+
+func (dl *DryLogger) AddImageStreamTag(ist *imageapi.ImageStreamTag) {
+	if tag := ist.Tag; tag != nil {
+		if tag.From != nil {
+			ist.Tag.From.Name = trimSHA256(ist.Tag.From.Name)
+		}
+	}
+	dl.AddObject(ist.DeepCopyObject())
+}
+
+func trimSHA256(value string) string {
+	reg := regexp.MustCompile("@sha256:[0-9a-f]{64}")
+	return reg.ReplaceAllString(value, "")
+}
+
+// GetObjects returns the list objects.
 func (dl *DryLogger) GetObjects() []runtime.Object {
 	dl.RLock()
 	defer dl.RUnlock()
 	return dl.objects
 }
 
+// Log encode/decode the objects and print them as a valid JSON array.
+func (dl *DryLogger) Log() error {
+	var errors []error
+	objects := dl.GetObjects()
+	for _, object := range objects {
+		var b []byte
+		buffer := bytes.NewBuffer(b)
+
+		err := encoder.Encode(object, buffer)
+		if err != nil {
+			fmt.Printf("Unexpected encode error '%v'\n", err)
+			errors = append(errors, err)
+			continue
+		}
+
+		_, _, err = decoder.Decode(buffer.Bytes(), nil, object)
```

Comment:
What does the `meta.Accessor()` return without encode/decode?

Reply:
If we don't encode/decode the `Kind` and `APIVersion` will be missing.